zerver/views/user_settings.py

```diff
@@ -25,7 +25,6 @@
 from zerver.lib.request import JsonableError
 from zerver.models import UserProfile, Realm, name_changes_disabled
 
-@authenticated_json_post_view
```

Comment:
You probably want to make something like `authenticated_json_patch_view` otherwise we can get a GET request against this view.